packages/material-ui/src/ButtonBase/ButtonBase.js

```diff
@@ -260,7 +262,7 @@ const ButtonBase = React.forwardRef(function ButtonBase(props, ref) {
       className={clsx(
         classes.root,
         {
-          [classes.disabled]: disabled,
+          [classes.disabled]: disabledProp,
```

Comment:
I had to introduce a private loading prop to disable the interaction without applying the disabled CSS. I don't like that, but I can't think of a better approach.